lightkurve/prf/scenemodels.py

```diff
@@ -0,0 +1,670 @@
+        self.params = self.get_initial_guesses()
+
+    def __repr__(self):
+        out = super(SceneModel, self).__repr__() + '\n'
+        out += '  Star priors:\n'+''.join(['    {}\n'.format(star) for star in self.star_priors])
+        out += '  Background prior:\n    {}\n'.format(self.background_prior)
+        out += '  Focus prior:\n    {}\n'.format(self.focus_prior)
+        out += '  Motion prior:\n    {}\n'.format(self.motion_prior)
+        out += '  PRF model:\n    {}\n'.format(self.prfmodel)
+        out += '  Options:\n    fit_background={}, fit_focus={}, fit_motion={}\n'.format(
+                        self.fit_background, self.fit_focus, self.fit_motion)
+        return out
+
+    def get_initial_guesses(self):
+        """Returns the prior means which can be used to initialize the model.
+
+        The guesses are obtained by taking the means of the priors.
+        """
+        initial_star_guesses = []
+        for star in self.star_priors:
+            initial_star_guesses.append(StarParameters(col=star.col.mean[0],
+                                                       row=star.row.mean[0],
+                                                       flux=star.flux.mean[0]))
+        background = BackgroundParameters(flux=self.background_prior.flux.mean[0],
+                                          fitted=self.fit_background)
+        focus = FocusParameters(scale_col=self.focus_prior.scale_col.mean[0],
+                                scale_row=self.focus_prior.scale_row.mean[0],
+                                rotation_angle=self.focus_prior.rotation_angle.mean[0],
+                                fitted=self.fit_focus)
+        motion = MotionParameters(shift_col=self.motion_prior.shift_col.mean[0],
+                                  shift_row=self.motion_prior.shift_row.mean[0],
+                                  fitted=self.fit_motion)
+        initial_params = SceneModelParameters(stars=initial_star_guesses,
+                                              background=background,
+                                              focus=focus,
+                                              motion=motion)
+        return initial_params
+
+    def predict(self, params=None):
+        """Returns a synthetic Kepler image given a set of scene parameters.
+
+        Attributes
+        ----------
+        params : ```SceneModelParameters``` object
+            Parameters which define the scene.
+
+        Returns
+        -------
+        synthetic_image : 2D ndarray
+            Predicted image given the parameters.
+        """
+        if params is None:
+            params = self.get_initial_guesses()
+        star_images = []
+        for star in params.stars:
+            star_images.append(self.prfmodel(flux=star.flux,
+                                             center_col=star.col + params.motion.shift_col,
+                                             center_row=star.row + params.motion.shift_row,
```

Comment:
Is ``star.col + params.motion.shift_col`` necessary? We want the center of the star relative to the TPF, I don't see why we need ``a + b`` to get that. Is this to plug in ``poscorr``? But in the case we want to fit both, aren't these gonna be degenerate?

Reply:
Oh, I see here, ``params.motion.shift_col`` is shared among all stars. That looks sweet!! =D

Reply:
Still, I don't quite understand this for the case of a single star.

Reply:
Yes, `shift_col` and `shift_row` model the spacecraft motion.  The idea is that we always use very narrow priors on the star positions (because we know star positions very well), and absorb spacecraft motion in `shift_col` and `shift_row`.  This allows us to prevent stars from moving relative to eachother in different cadences, which we know can't happen.

In the case of a single star the user can choose to keep either the motion or the star position fixed.  It is very easy to do this in the new API by setting `fit_motion=False` in the `SceneModel`.

Reply:
Makes a lot of sense =)